pkg/daemon/api/object.go

```diff
@@ -154,7 +154,7 @@ func (h *Handler) GetObjectStoreConnectionInfo(w http.ResponseWriter, r *http.Re
 	}
 
 	s3Info := &model.ObjectStoreConnectInfo{
-		Host:      oprgw.InstanceName(storeName),
+		Host:      fmt.Sprintf("%s.%s", oprgw.InstanceName(storeName), h.config.namespace),
```

Comment:
We are already generating a service, which is already resolvable via DNS. https://github.com/rook/rook/blob/master/pkg/operator/object/ceph/rgw.go#L332

So why not use that instead of recreating a new domain name?